### ISSUE TITLE
DTLS Session Resumption doesn't consider SNI 

### ISSUE THREAD

**Issue by boaks** [eclipse/californium#644]
Branch 2.0.x:

RFC6066, section 4, page 7:

> A server that implements this extension MUST NOT accept the request
> to resume the session if the server_name extension contains a
> different name. Instead, it proceeds with a full handshake to
> establish a new session. When resuming a session, the server MUST
> NOT include a server_name extension in the server hello.
> 

Test added to `DTLSConnectorTest`:

```
	@Test
	public void testConnectorPerformsFullHandshakeWhenResumingWithDifferentSni() throws Exception {
		// Do a first handshake
		RawData raw = RawData.outbound("Hello World".getBytes(), new AddressEndpointContext(serverEndpoint, "server.one", null), null, false);
		givenAnEstablishedSession(raw, true);
		byte[] sessionId = establishedServerSession.getSessionIdentifier().getId();

		// Force a resume session the next time we send data
		client.forceResumeSessionFor(serverEndpoint);
		Connection connection = clientConnectionStore.get(serverEndpoint);
		assertArrayEquals(sessionId, connection.getEstablishedSession().getSessionIdentifier().getId());
		client.start();

		// Prepare message sending
		final String msg = "Hello Again";
		CountDownLatch latch = new CountDownLatch(1);
		clientRawDataChannel.setLatch(latch);

		// send message
		RawData data = RawData.outbound(msg.getBytes(), new AddressEndpointContext(serverEndpoint, "server.two", null), null, false);
		client.send(data);
		assertTrue(latch.await(MAX_TIME_TO_WAIT_SECS, TimeUnit.SECONDS));

		// check session id was not equals
		connection = clientConnectionStore.get(serverEndpoint);
		Assert.assertThat(sessionId, not(equalTo(connection.getEstablishedSession().getSessionIdentifier().getId())));
		assertClientIdentity(RawPublicKeyIdentity.class);
	}
```

fails, so the session is resumed without considering the changed SNI.

**Comment by boaks** [comment]
Using SNI (provide a hostname instead of the ip-address as destination), a resumption fails always!
The Resuming DTLSSession has not virtual name (it's only set for new sessions), therefore the matcher fails.

**Comment by boaks** [comment]
Fixed with PR #816
